File: 백준/Silver/2606. 바이러스/바이러스.py
from collections import deque

n = int(input())
v = int(input())

graph = [[] for _ in range(n+1)]
visited = [False] * (n+1)

for i in range(v):
    a,b = map(int, input().split())
    graph[a] += [b]
    graph[b] += [a]

visited[1] = True
queue = deque()
queue.append(1)

while queue:
    index = queue.popleft()
    for i in graph[index]:
        if not visited[i]:
            visited[i] = True
            queue.append(i)
            
print(sum(visited)-1)

File: 백준/Silver/2606. 바이러스/바이러스.swift
let computerNum = Int(readLine()!)!
let linkComputerNum = Int(readLine()!)!

var graph = [[Int]](repeating: [], count: computerNum + 1)
var visited = [Bool](repeating: false, count: computerNum + 1)

for _ in 0..<linkComputerNum {
    let input = readLine()!.split(separator: " ").map { Int($0)! }
    
    graph[input[0]].append(input[1])
    graph[input[1]].append(input[0])
}

print(dfs(1) - 1)

func dfs(_ start: Int) -> Int {
    var count = 1
    visited[start] = true
    
    for i in graph[start] {
        if visited[i] == false {
            count += dfs(i)
        }
    }
    
    return count
}